test/VaultUSDTDeploy.t.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: LicenseRef-SkyAlpha-Proprietary
+// © 2025 SkyAlpha Ventures LLC. All rights reserved. Use subject to LICENSE.txt.
+// No claims against contributors: to the maximum extent permitted by applicable law, each contributor
+// provides its contributions "AS IS", disclaims all warranties, and shall have no liability whatsoever
+// for any damages arising from or relating to the Software or its use.
+
+pragma solidity ^0.8.18;
+
+import {Test, console} from "forge-std/Test.sol";
+import {IPool} from "ajna-core/interfaces/pool/IPool.sol";
+import {PoolInfoUtils} from "ajna-core/PoolInfoUtils.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import {Vault} from "../src/Vault.sol";
+import {Buffer} from "../src/Buffer.sol";
+import {IBuffer} from "../src/interfaces/IBuffer.sol";
+import {IVault} from "../src/interfaces/IVault.sol";
+import {VaultAuth, IVaultAuth} from "../src/VaultAuth.sol";
+import {ERC4626} from "../src/ERC4626.sol";
+
+/**
+ * @title VaultUSDTDeployTest
+ * @notice Test to reproduce USDT vault deployment failure on mainnet fork
+ * @dev This test uses mainnet fork to test deployment with:
+ *      - USDT (0xdac17f958d2ee523a2206206994597c13d831ec7) as quote token
+ *      - wstETH/USDT pool (0xd7fef7e3ac0440086f6322dc47d72e6c96caa6ca) as underlying pool
+ */
+contract VaultUSDTDeployTest is Test {
+
+    // Mainnet addresses
+    address public constant USDT = 0xdAC17F958D2ee523a2206206994597C13D831ec7;
+    address public constant WSTETH_USDT_POOL = 0xD7feF7E3aC0440086f6322dC47d72E6C96caA6cA;
+    address public constant AJNA_INFO = 0x30c5eF2997d6a882DE52c4ec01B6D0a5e5B4fAAE;
+
+    // Test actors
+    address public deployer = makeAddr("deployer");
+
+    // Contracts
+    Vault public vault;
+    VaultAuth public auth;
+    Buffer public buffer;
+    IPool public pool;
+    PoolInfoUtils public info;
+
+    function setUp() public {
+        // Fork mainnet at a recent block
+        string memory rpcUrl = vm.envString("ETH_RPC_URL");
+        vm.createSelectFork(rpcUrl);
+
+        console.log("=== Forked Mainnet ===");
+        console.log("Block number:", block.number);
+        console.log("Block timestamp:", block.timestamp);
+        console.log("");
```

Comment:
The current test suite allows running with ETH_RPC_URL and locally with mocks.  Can you add the `onlyLiveFork` modifier to these so the local test suite can still run?